src/middleware/subscribeWithSelector.ts

```diff
@@ -10,6 +10,31 @@ import {
   Subscribe,
 } from '../vanilla'
 
+declare module '../vanilla' {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  interface StoreMutators<S, A> {
+    ['zustand/subscribeWithSelector']: WithSelectorSubscribe<S>
+  }
+}
+
+type WithSelectorSubscribe<S> = S extends { getState: () => infer T }
+  ? S & StoreSubscribeWithSelector<Extract<T, State>>
+  : never
+
+interface StoreSubscribeWithSelector<T extends State> {
+  subscribe: <U>(
+    selector: (state: T) => U,
+    listener: (selectedState: U, previousSelectedState: U) => void,
+    options?: {
+      equalityFn?: (a: U, b: U) => boolean
+      fireImmediately?: boolean
+    }
+  ) => () => void
+}
+
+/**
+ * @deprecated use `Mutate<StoreApi<T>, ["zustand/subscribeWithSelector"]>`
```

Comment:
Can you modify middlewareTypes.test.tsx to show the usage of `Mutate`? Just removing works?

Reply:
Yep sure will do

Reply:
Hmm the module augmentation seems to be not working, looking into it.

Reply:
Oh lol, it should be `Mutate<StoreApi<T>, [["zustand/subscribeWithSelector", never]]>` instead of `Mutate<StoreApi<T>, ["zustand/subscribeWithSelector"]>` fixed it.

Thinking out loud: We could add a sugar to make `Mutator` same as `[Mutator, never]` ie if the payload is not specified it's `never`. But let's not complicate things already, will think about hit later.

Reply:
I think I get what you mean. It's a bit complicated, but DX-wise, it should be much better. `, never` is just too confusing.

Reply:
I kinda agree, but then I wonder if it'll be complicated if a middleware has a payload eg `Mutate<Store<T>, ["zustand/subscribeWithSelector", ["zustand/persist", Partial<T>]]>` I worry this inconsistency in shape would be even more confusing. I think we should keep it as it is.

Reply:
I see. yeah, that works.